### DISCUSSION TITLE
How to clear query history

### DISCUSSION THREAD

**Discussion by cereda** [tconbeer/harlequin#593] [Q&A, Troubleshooting, Support] [answered]
Hi friends! First of all, thanks for the amazing application! :tada: 

Hopefully the title says it all: I was wondering how to clear query history for a certain `.duckdb` database. I tried looking for any history file in the most common directories (e.g, `~/.config`), but could not find anything relevant. Also, my last query stays there in the editor — I would love to have a way to clear this data and start afresh on every run, if possible. :wink: 

Any suggestion is appreciated! _(Also, apologies in advance if I missed something obvious in the documentation...)_

Cheerio!

**Comment by cereda** [comment, marked as answer]
Found it! :wink: It's just a matter of cleaning up `~/.cache/harlequin` (there are two `.pickle` files in there). Thanks again and sorry for the noise!

**Reply by tconbeer** [reply]
Yes, we use the platform-specific "user cache directory" to store the data catalog, open buffers, and query history for each connection.
